lib/cartopy/mpl/ticker.py

```diff
@@ -64,6 +64,9 @@ def __call__(self, value, pos=None):
                                        ccrs.Mercator)):
                 raise TypeError("This formatter cannot be used with "
                                 "non-rectangular projections.")
+            if source.globe != self._target_projection.globe:
+                # The transforms need to use the same globe
+                self._target_projection = ccrs.PlateCarree(globe=source.globe)
```

Comment:
This is the only part that gives me pause. I'm not really sure what's going on here.

Reply:
That is because it is initialized as a class variable above without a globe: https://github.com/SciTools/cartopy/blob/0d584493a033df2bbde4318ca97fb266249c986d/lib/cartopy/mpl/ticker.py#L23
then we try to do the transform here in the call and it raises if the globes are different. But, we can't know what the source formatter is until this point, and we don't want to create a new projection every call either I don't think, so I added this check for that case.

Reply:
Could we implement `set_axis` (which exists from `Formatter`->`TickHelper`) to set what we need at that time?

Reply:
That is a good idea. I did that in a separate commit so it is easier to see the update. All of the tests we have currently call `.axis = Mock(...)`, which avoids the `set_axis()` path. So, do we think that anyone would call `.axis = ...` on the formatter and thus we should keep the logic in the __call__ method instead... Or I guess we could turn `.axis` into a property?

Reply:
Given that `.axis` and `set_axis()` are upstream APIs, I think it makes the most sense for any notion of `axis` being a property to be handled there.

I feel comfortable not worrying about users manually setting `.axis`; my advice in that case would be for users to use `set_axis` too.